x/staking/keeper/msg_server.go

```diff
@@ -203,6 +155,10 @@ func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*typ
 		return nil, sdkerrors.Wrapf(types.ErrBadDenom, "got %s, expected %s", msg.Amount.Denom, bondDenom)
 	}
 
+	// Thoughts: since delegation goes to Unbonded pool, as long as we only run validator set update on epochs, it won't affect
```

Comment:
This is different than MsgCreateValidator.  It was okay in MsgCreateValidator because the validator could not have been bonded already, and so would not be gaining rewards or anything.

However, in a Delegate msg, I could be delegating to a validator who is in the active set, and even if we don't update the validator set until the end of the epoch, by executing the Delegation immediately, I will start getting rewards and slashes of the validator, which is not what we want